IO files

emaximili opened the pull request:
Fixes #16
Fixes #19

emaximili committed:
Added possibility to import / export files. 
- simple button at the bottom. 
- converted all the old examples to new input files.
- fixed visualization of inluence line bug.

emaximili committed:
removed old input files from git

federicobetti-ethz reviewed (commented):
Good job! One question: do we really need all these .mat files as examples? I.e. are they really representative or could we just use say 4-5?

In general, I would say before merging it, let's take a look at it together on Wednesday, unless you tell me not merging it before that somehow blocks your progress.

emaximili committed:
Adjusted loading and saving buttons. Made info button blue. Changed beam load names.
Fixed static condensation logic:
- table on off works
- Multiple nodes now supported
- Results shown in workspace

emaximili committed:
Added error VAlidation upon switching panels:
- Separate functions for each panel
- Same checks as in ValidateAnalysisModel but adjusted for different input type

Put setup file in root directory

federicobetti-ethz commented:
Hello @emaximili,

Thanks for reaching out. Yes, as we agreed on Wednesday no worries if you took care of the error handling on this branch as well.
I will take a look now at the actual code, however would you be available for a quick Zoom call where you show me what you did directly on the GUI? We can do it tomorrow morning first thing if you wish.

federicobetti-ethz commented on src/GUI/Validation/validateBeam.m:
```diff
@@ -0,0 +1,44 @@
+        'Tabelle "Stäbe": Spalte "StartKnoten" verweist auf ungültige Knoten (Zeile(n): %s).', ...
+        num2str(rows(:).'));
+end
+if any(badEnd)
```
these validate* functions are the ones called when switching tab I assume, right?

emaximili replied:
Yes exactly. I can show it to you tomorrow.

federicobetti-ethz commented on src/GUI/Validation/validateMaterial.m:
```diff
@@ -0,0 +1,34 @@
+        ' Bitte einen gültigen Namen eingeben.'], ...
+        strtrim(num2str(badRows.')));
+end
+
```
a string input for the E Modulus will automatically be changed to Nan, so this is covered as well, can you confirm?

emaximili replied:
Yes exactly.

federicobetti-ethz reviewed (commented):
A few comments here and there, but again I need to see it on the GUI before approving.

> Another small comment: the folder scripts contains only one file, so you can move it outside and remove the folder. It is also nicer to see the setup.m at the first level.

PS: it does not look like this has been addressed by the way, if so please do it

emaximili committed:
Mbeautified drawOriginalFig

federicobetti-ethz commented on src/DSM/Zeichnen/drawOriginalFig.m:
```diff
@@ -252,6 +275,14 @@
         eKL = SKKord + R' * eKordLine;
 
```
Thank you, I will resolve.

emaximili committed:
Added functionality to delete all table input. Improved the error handling.

emaximili committed:
added icons to git

federicobetti-ethz reviewed (approved):
Good to go from my side.
